ext/drx/drx.c

```diff
@@ -129,13 +137,20 @@ bool
 drx_init(void)
 {
     /* drx_insert_counter_update() needs 1 slot on x86 plus the 1 slot
-       drreg uses for aflags, and 2 reg slots on aarch, so 2 on both.
+     * drreg uses for aflags, and 2 reg slots on aarch, so 2 on both.
+     * drx_expand_scatter_gather() needs 4 slots in app2app phase, which
+     * cannot be reused by other phases. So, ideally we should reserve
+     * 6 slots for drx. But we settle with 4 to avoid stealing too many
+     * slots from other clients/libs. When drx needs more for instrumenting
+     * scatter/gather instrs, we fall back on DR slots. As scatter/gather
+     * instrs are spilt into their own bbs, this effect will be limited.
+     * On Windows however we reserve even fewer slots, as they are
+     * shared with the application and reserving even one slot can result
+     * in failure to initialize for certain applications (e.g. i#1163).
      * We set do_not_sum_slots to true so that we only ask for *more* slots
-     * if the client doesn't ask for any. Another drreg_init() call is made
-     * in case drx_expand_scatter_gather() is called. More slots are reserved
-     * in that case.
+     * if the client doesn't ask for any.
```

Comment:
With do_not_sum_slots still set to true, this PR is not doing what it says in this comment: it is not reserving 4 slots separate from the client.  If the client itself wants 4 this PR is doing nothing; if the client wants 3 this PR is reserving just 1 extra.

Reply:
Fixed it. Thanks for catching.